### PR TITLE
Add colorblind mode

### PR DESCRIPTION
**Pull request by lauble**

### CODE REVIEW

**Comment by lauble** [comment]
#1213

**Comment by lauble** [comment]
#### Description
Implemented a colorblind mode option for the Rapid map. This PR aims to resolve Issue #1213.

These changes will help improve map accessibility for people with red-green colorblindness. It also sets the stage for future colorblind modes to be added. The modifications include the addition of a new color scheme, `red_blind`, or Protanopia, and a map color scheme combobox in the preferences sidebar that changes the map color scheme based on the user's selection.

#### Screenshots:

#### Code weaknesses and future updates:

**Comment by lauble** [comment]
@Bonkles adding you here for progress updates and any feedback you may have.

**Comment by lauble** [comment]
Screenshot 2023-11-28 at 2 16 01 PM

Are these run fail errors problematic?

**Comment by Bonkles** [comment]
Let me look into what's going on with the infra errors- these are not your fault @lauble !

**Comment by lauble** [comment]
@Bonkles I need some help with the combobox labels. I've been trying to get the info from the `core.yaml` file, but it isn't showing the information I want in the UI. (i.e. I want `default` to show as `Default`, `red_blind` to show as `Protanopia`, etc.) Can you help point me in the right direction?

**Comment by Bonkles** [comment]
I'd say to check to see if `d` really is what you think it is in this case- since the combo box data values are objects with a `title` and a `value` I would think you'd need to specify either `d.title` or `d.value` like so: 

`.text(d => l10n.t(`preferences.color_selection.${d.title}`))`

File: modules/ui/sections/color_selection.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  let comboData = [];
+
+  function loadComboBoxData(){
+    let colorSchemeKeys = Object.keys(colors.getAllColorSchemes());
+
+    for (let i = 0; i < colorSchemeKeys.length; i++) {
+      let colorObject = {};
+      let colorSchemeTitle = colorSchemeKeys[i];
+      colorObject.title = colorSchemeTitle;
+      colorObject.value = colorSchemeTitle;
```
Comment (Bonkles) [inline modules/ui/sections/color_selection.js:21]: This function here is where you should put your string resource fetch: I believe it is the `values` that are actually displayed by the comboBox code, so change the `value` of each object to essentially the same l10n call you have further down. 

```colorObject.value = l10n.t(`preferences.color_selection.${colorSchemeTitle}`);```

File: modules/ui/sections/color_selection.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        .title(l10n.tHtml('preferences.color_selection.tooltip')));
+
+
+    let pickerCombo = colorOptionsListEnter.append('div');
+    let options = ['default', 'red_blind', 'green_blind'];
+
+    pickerCombo.append('input')
+    .attr('class', 'color-select')
+      .attr('placeholder', l10n.t('preferences.color_selection.placeholder'))
+      .data(options)
```
Comment (Bonkles) [inline modules/ui/sections/color_selection.js:54]:
```suggestion
```

File: modules/ui/sections/color_selection.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            context.systems.map.deferredRedraw();
+          }
+
+        } else {
+          d3_event.currentTarget.value = '';
+          _colorSelectedId = null;
+        }
+      });
+
+    colorCombo.data(comboData);
```
Comment (Bonkles) [inline modules/ui/sections/color_selection.js:77]: no changes necessary here, just set a breakpoint here and you should see the `comboData` has something like 

`{ title: 'default', value: 'Default'}, `{title:` 'red_blind', value: 'Protanopia'} etc.

Comment (lauble) [reply modules/ui/sections/color_selection.js:75]: 😄 
Screenshot 2023-12-04 at 12 09 35 PM

File: modules/ui/sections/color_selection.js (hunk 1 above)
Comment (lauble) [reply modules/ui/sections/color_selection.js]: wunderbar

Screenshot 2023-12-04 at 12 11 42 PM

**Comment by bhousel** [comment]
> These changes emulate color blindness, rather than correct for it. Future research on the use of color matrices to correct for color blindness is needed.

Thanks, this looks like a good start.  That's a good point that simulating color blindness isn't really what we're after, but correcting for it could be useful.

This PR has gone a bit stale and I need to fix some color styling issues today, so I'm going to merge as-is and hide/comment some of the parts of the code until it can be developed more fully.

**Pull request merged**